charts/tempo-distributed/templates/admin-api/_helpers.tpl

```diff
@@ -0,0 +1,7 @@
+{{/*
+adminApi imagePullSecrets
+*/}}
+{{- define "tempo.adminApiImagePullSecrets" -}}
+{{- $dict := dict "tempo" .Values.tempo.image "component" .Values.adminApi.image "global" .Values.global.image -}}
+{{- include "tempo.adminApiImagePullSecrets" $dict -}}
```

Comment:
Is this `include` desired here?  It looks to me like we are defining a function and then recursively calling it.  I suspect we want the same approach to all the components attempting to pull images from private registries, and so creating one template per component seems a little verbose.  Can we unify the templating?

Reply:
Based on `templates/distributor`/helpers-distributor.tpl` it looks like we can change this include line to point to the top level include `tempo.imagePullSecrets`

```
{{/*
distributor imagePullSecrets
*/}}
{{- define "tempo.distributorImagePullSecrets" -}}
{{- $dict := dict "tempo" .Values.tempo.image "component" .Values.distributor.image "global" .Values.global.image -}}
{{- include "tempo.imagePullSecrets" $dict -}}
{{- end }}
```

Reply:
Yep that was my bad there. Thanks for that. I'm also looking at updating the `tokengenJob` to support this too.